Objects/typeobject.c

```diff
@@ -343,6 +344,49 @@ _PyStaticType_GetBuiltins(void)
 
 /* end static builtin helpers */
 
+static void
+type_set_flags(PyTypeObject *tp, unsigned long flags)
+{
+#ifdef Py_GIL_DISABLED
+    if (tp->tp_flags & Py_TPFLAGS_READY) {
```

Comment:
Shouldn't the read operation be atomic as well?

Reply:
I don't think so.  This is used either if the `TYPE_LOCK` is held or if the type is newly created and the lock will be held later, before exposing the type to other threads.  So there is no way for another thread to write to the flags.